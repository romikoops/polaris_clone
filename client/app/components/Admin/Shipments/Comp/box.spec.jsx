import * as React from 'react'
import { shallow } from 'enzyme'
import { theme } from '../../../../mock'

import AdminShipmentsBox from './box'

const propsBase = {
  shipments: [],
  handleClick: null,
  seeAll: null,
  theme,
  confirmShipmentData: {},
  userView: false,
  page: 1,
  nextPage: null,
  prevPage: null,
  handleSearchChange: null,
  numPages: 1
}

test('shallow render', () => {
  expect(shallow(<AdminShipmentsBox {...propsBase} />)).toMatchSnapshot()
})

test('page > 10', () => {
  const props = {
    ...propsBase,
    page: 11
  }
  expect(shallow(<AdminShipmentsBox {...props} />)).toMatchSnapshot()
})

test('shipments is truthy', () => {
  const props = {
    ...propsBase,
    shipments: [{}]
  }
  expect(shallow(<AdminShipmentsBox {...props} />)).toMatchSnapshot()
})
