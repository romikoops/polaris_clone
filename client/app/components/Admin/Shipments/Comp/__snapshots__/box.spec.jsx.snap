// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`page > 10 1`] = `
<div
  className="layout-row flex-100
         layout-wrap layout-align-start-center searchable"
>
  <div
    className="flex-100 layout-row layout-align-end-center searchable_header"
  >
    <div
      className="input_box flex-40 layout-row layout-align-end-center"
    >
      <input
        name="search"
        onChange={null}
        placeholder="Search shipments"
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    className="flex-95 flex-offset-5 layout-row layout-wrap layout-align-start-center margin_bottom"
  >
    <div
      className="flex-100 layout-row layout-align-space-between-center sec_subheader"
    >
      <p
        className=" sec_subheader_text flex-none"
      >
         No Shipments yet
      </p>
    </div>
    <p
      className="flex-none"
    >
       As shipments are requested, they will appear here
    </p>
  </div>
  <div
    className="flex-95 layout-row layout-align-center-center margin_bottom"
  >
    <div
      className="
                      flex-15 layout-row layout-align-center-center pointy
                      navigation_button 
                    "
      onClick={null}
    >
      <i
        className="fa fa-chevron-left"
      />
      <p>
            
        Back
      </p>
    </div>
    <p>
      11
    </p>
    <div
      className="
                      flex-15 layout-row layout-align-center-center pointy
                      navigation_button disabled
                    "
      onClick={null}
    >
      <p>
        Next
            
      </p>
      <i
        className="fa fa-chevron-right"
      />
    </div>
  </div>
</div>
`;

exports[`shallow render 1`] = `
<div
  className="layout-row flex-100
         layout-wrap layout-align-start-center searchable"
>
  <div
    className="flex-100 layout-row layout-align-end-center searchable_header"
  >
    <div
      className="input_box flex-40 layout-row layout-align-end-center"
    >
      <input
        name="search"
        onChange={null}
        placeholder="Search shipments"
        type="text"
        value=""
      />
    </div>
  </div>
  <div
    className="flex-95 flex-offset-5 layout-row layout-wrap layout-align-start-center margin_bottom"
  >
    <div
      className="flex-100 layout-row layout-align-space-between-center sec_subheader"
    >
      <p
        className=" sec_subheader_text flex-none"
      >
         No Shipments yet
      </p>
    </div>
    <p
      className="flex-none"
    >
       As shipments are requested, they will appear here
    </p>
  </div>
  <div
    className="flex-95 layout-row layout-align-center-center margin_bottom"
  >
    <div
      className="
                      flex-15 layout-row layout-align-center-center pointy
                      navigation_button disabled
                    "
      onClick={null}
    >
      <i
        className="fa fa-chevron-left"
      />
      <p>
            
        Back
      </p>
    </div>
    <p>
      1
    </p>
    <div
      className="
                      flex-15 layout-row layout-align-center-center pointy
                      navigation_button disabled
                    "
      onClick={null}
    >
      <p>
        Next
            
      </p>
      <i
        className="fa fa-chevron-right"
      />
    </div>
  </div>
</div>
`;

exports[`shipments is truthy 1`] = `
<div
  className="layout-row flex-100
         layout-wrap layout-align-start-center searchable"
>
  <div
    className="flex-100 layout-row layout-align-end-center searchable_header"
  >
    <div
      className="input_box flex-40 layout-row layout-align-end-center"
    >
      <input
        name="search"
        onChange={null}
        placeholder="Search shipments"
        type="text"
        value=""
      />
    </div>
  </div>
  <ShipmentOverviewCard
    admin={true}
    confirmShipmentData={Object {}}
    hubs={Object {}}
    noTitle={true}
    shipments={
      Array [
        Object {},
      ]
    }
    t={[Function]}
    theme={
      Object {
        "colors": Object {
          "primary": "#333",
          "secondary": "#fafafa",
        },
      }
    }
  />
  <div
    className="flex-95 layout-row layout-align-center-center margin_bottom"
  >
    <div
      className="
                      flex-15 layout-row layout-align-center-center pointy
                      navigation_button disabled
                    "
      onClick={null}
    >
      <i
        className="fa fa-chevron-left"
      />
      <p>
            
        Back
      </p>
    </div>
    <p>
      1
    </p>
    <div
      className="
                      flex-15 layout-row layout-align-center-center pointy
                      navigation_button disabled
                    "
      onClick={null}
    >
      <p>
        Next
            
      </p>
      <i
        className="fa fa-chevron-right"
      />
    </div>
  </div>
</div>
`;
